tests/functional/iac/test_iac_scan_diff.py

```diff
@@ -87,6 +120,43 @@ def test_iac_scan_diff_new_vuln(tmp_path: Path) -> None:
     assert "file2.tf" in result.stdout
 
 
+def test_iac_scan_diff_new_vuln_inner_dir(tmp_path: Path) -> None:
+    # GIVEN a git repository
+    repo = Repository.create(tmp_path)
+    repo.create_commit()
+
+    # AND inner directory
+    inner_dir_path = tmp_path / "innerdir"
+    inner_dir_path.mkdir()
+
+    # AND a first commit with a vulnerability in file1.tf
+    file1 = inner_dir_path / "file1.tf"
+    file1.write_text(_IAC_SINGLE_VULNERABILITY)
+    repo.add(file1)
+    repo.create_commit()
+
+    # AND a second commit with another vulnerability in file2.tf
+    file2 = inner_dir_path / "file2.tf"
+    file2.write_text(_IAC_SINGLE_VULNERABILITY)
+    repo.add(file2)
+    repo.create_commit()
+
+    # WHEN scanning the diff between current and HEAD~1
+    # (meaning reference should be one commit behind)
+    args = ["diff", "--ref", "HEAD~1", "--verbose", str(inner_dir_path)]
+    # THEN exit code is 1 (new vuln detected)
+    result = run_ggshield_iac_scan(*args, cwd=inner_dir_path, expected_code=1)
+
+    # AND vulnerability of file1.tf shows as unchanged
+    # AND vulnerability of file2.tf shows as new
+    assert "0 incidents deleted" in result.stdout
+    assert "1 incident remaining" in result.stdout
+    assert "1 new incident detected" in result.stdout
+    # AND details about the new vulnerability are shown
+    assert "file2.tf" in result.stdout
+    assert "The incident was found between" not in result.stdout
```

Comment:
Maybe we should test the presence of parts of the `_IAC_SINGLE_VULNERABILITY` content ? That would be more robust, in particular if we change the wording of the fallback message.

Reply:
And same below